great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1071 @@
+                #                     f"{semantic_type} expectation, but ensure that this is correctly configured.")
+                # else:
+                #     logger.debug(f"Semantic_type: {semantic_type} is unknown. Skipping")
+        return self.semantic_types_dict
+
+    def _add_column_type_to_column_info(self, dataset, column_name):
+        """
+        Adds the data type of a column to the column_info dictionary on self
+        Args:
+            dataset: A GE dataset
+            column_name: The name of the column for which to retrieve the data type
+
+        Returns:
+            The type of the column
+        """
+        if "expect_column_values_to_be_in_type_list" in self.excluded_expectations:
+            logger.debug(
+                "expect_column_values_to_be_in_type_list is in the excluded_expectations list. This"
+                "expectation is required to establish column data, so it will be run and then removed from the"
+                "expectation suite."
+            )
+        column_info_entry = self.column_info.get(column_name)
+        if not column_info_entry:
+            column_info_entry = {}
+            self.column_info[column_name] = column_info_entry
+        column_type = column_info_entry.get("type")
+        if not column_type:
+            column_type = self._get_column_type(dataset, column_name)
+            column_info_entry["type"] = column_type
+
+            dataset.set_config_value("interactive_evaluation", True)
+
+        return column_type
+
+    def _get_column_type(self, dataset, column):
+        """
+        Determines the data type of a column by evaluating the success of `expect_column_values_to_be_in_type_list`.
+        In the case of type Decimal, this data type is returned as NUMERIC, which contains the type lists for both INTs
+        and FLOATs.
+
+        The type_list expectation used here is removed, since it will need to be built once the build_suite function is
+        actually called. This is because calling build_suite wipes any existing expectations, so expectations called
+        during the init of the profiler do not persist.
+
+        Args:
+            dataset: A GE dataset
+            column: The column for which to get the data type
+
+        Returns:
+            The data type of the specified column
+        """
+        # list of types is used to support pandas and sqlalchemy
+        type_ = None
+        dataset.set_config_value("interactive_evaluation", True)
+        try:
+
+            if (
+                dataset.expect_column_values_to_be_in_type_list(
+                    column, type_list=sorted(list(ProfilerTypeMapping.INT_TYPE_NAMES))
+                ).success
+                and dataset.expect_column_values_to_be_in_type_list(
+                    column, type_list=sorted(list(ProfilerTypeMapping.FLOAT_TYPE_NAMES))
+                ).success
+            ):
+                type_ = "NUMERIC"
```

Comment:
You might want to use the enum here and elsewhere that these type and cardinality strings are used e.g. 
```suggestion
                type_ = ProfilerDataType.NUMERIC.value
```
This way if the string changes, we only have to change it in the enum. Especially since these names are pretty generic, it may be difficult to do a find and replace.